zeebe/gateway-grpc/src/main/java/io/camunda/zeebe/gateway/interceptors/impl/AuthenticationHandler.java

```diff
@@ -75,17 +78,61 @@ public Either<Status, Context> authenticate(final String authorizationHeader) {
       final var username =
           token.getClaims().get(oidcAuthenticationConfiguration.getUsernameClaim());
 
-      if (username == null) {
-        return Either.left(
-            Status.UNAUTHENTICATED.augmentDescription(
-                "Expected a claim '%s' in the token, but it was not present"
-                    .formatted(oidcAuthenticationConfiguration.getUsernameClaim())));
+      if (username != null) {
+        if (username instanceof String) {
```

Comment:
:wrench: `instanceof` is already an implicit null check

Reply:
As discussed via Slack, we need this in here to be able to handle the different variations of error handling so for now I'll leave it in